base/sparse/sparsematrix.jl

```diff
@@ -3007,3 +3009,28 @@ end
 (+)(A::SparseMatrixCSC, J::UniformScaling) = A + J.λ * speye(A)
 (-)(A::SparseMatrixCSC, J::UniformScaling) = A - J.λ * speye(A)
 (-)(J::UniformScaling, A::SparseMatrixCSC) = J.λ * speye(A) - A
+
+#set-like operations
+function union!{Tv,Ti<:Integer}(s::Set,x::SparseMatrixCSC{Tv,Ti})
+    vals = nonzeros(x)
+    s = union!(s,vals)
+    if length(x) == length(vals)
```

Comment:
Yes, either the statement inside is wrong or the condition is wrong. Have fixed.